tests/conftest.py

```diff
@@ -103,6 +105,22 @@ def backend(request):
     await send({"type": "http.response.body", "body": b"Hello, world!"})
 
 
+async def server_crash(scope, receive, send):
+    crash_ms_str: str = scope["path"].replace("/server_crash/", "")
+    try:
+        crash_ms = float(crash_ms_str)
+    except ValueError:
+        crash_ms = 100
+    await asyncio.sleep(crash_ms / 1000.0)
```

Comment:
Is it important to wait for a bit before closing the connection prematurely? (Just wondering if we can simplify the failing test case even further)

Reply:
we probably could.  I simply wanted to avoid any chance of a race condition.